packages/jest-cli/src/generateEmptyCoverage.js

```diff
@@ -29,16 +32,15 @@ export default function(
     collectCoverageOnlyFrom: globalConfig.collectCoverageOnlyFrom,
   };
   if (Runtime.shouldInstrument(filename, coverageOptions, config)) {
-    // Transform file without instrumentation first, to make sure produced
-    // source code is ES6 (no flowtypes etc.) and can be instrumented
+    // Transform file with instrumentation to make sure initial coverage data is well mapped to original code.
     const transformResult = new Runtime.ScriptTransformer(
       config,
-    ).transformSource(filename, source, false);
-    const instrumenter = createInstrumenter();
-    instrumenter.instrumentSync(transformResult.code, filename);
+    ).transformSource(filename, source, true);
+    const extracted = readInitialCoverage(transformResult.code);
+
     return {
-      coverage: instrumenter.fileCoverage,
-      sourceMapPath: transformResult.sourceMapPath,
+      coverage: new FileCoverage(extracted.coverageData),
+      sourceMapPath: null,
```

Comment:
sourceMapPath should still be relevant here. Runtime.ScriptTransformer may call out to custom transformers (instead of babel-jest) that do a two-step pass (compile, then instrument) similar to what was happening here originally. Without a source map, we'd just have coverage data for compiled code, not source code. It's the way things currently work for non-empty coverage.